frontend/include/chpl/types/LoopExprIteratorType.h

```diff
@@ -73,6 +74,7 @@ class LoopExprIteratorType final : public IteratorType {
   bool contentsMatchInner(const Type* other) const override {
     auto rhs = (LoopExprIteratorType*) other;
     return yieldType_ == rhs->yieldType_ &&
+           poiScope_ == rhs->poiScope_ &&
```

Comment:
IMO it's better to have a function for IteratorType to call that handles the fields there. An example from uAST is `simpleBlockLikeContentsMatchInner`.